torch/csrc/autograd/custom_function.cpp

```diff
@@ -86,6 +86,84 @@ variable_list _wrap_outputs(const variable_list &input_vars,
 }
 
 
+template <typename T, typename... Args>
+void extract_vars(variable_list& list, T&& cur, Args&& ... args) {
+  extract_vars(list, std::forward<Args>(args)...);
+}
+
+template <typename... Args>
+void extract_vars(variable_list& list, Variable&& cur, Args&& ... args) {
+  list.push_back(cur);
+  extract_vars(list, std::forward<Args>(args)...);
+}
+
+template <typename... Args>
+void extract_vars(variable_list& list, Variable& cur, Args&& ... args) {
+  list.push_back(cur);
+  extract_vars(list, std::forward<Args>(args)...);
+}
+
+template <typename... Args>
+void extract_vars(variable_list& list, Args&& ... args) {
+}
 
 
+template<class T>
+template<typename... Args>
+variable_list CFunction<T>::apply(Args&&... args) {
+  variable_list input_vars;
+  extract_vars(input_vars, std::forward<Args>(args)...);
```

Comment:
I think you don't actually want a `std::forward` here; if you have an rvalue reference, you don't want to move it into `extract_vars` (TBF, no move will actually happen because you don't actually do anything with those arguments; it's more about the principle of the thing.)

Reply:
@smessmer maybe you have a better idea?